packages/core/admin/admin/src/pages/MarketplacePage/tests/plugins.test.js

```diff
@@ -140,12 +146,254 @@ describe('Plugins tab', () => {
+      // They should see the collection option results
+      const collectionPlugin = screen.getByText('Gatsby Preview');
+      const notCollectionPlugin = screen.queryByText('Comments');
+      expect(collectionPlugin).toBeVisible();
+      expect(notCollectionPlugin).toEqual(null);
+      // They should see the category option results
+      const categoryPlugin = screen.getByText('CKEditor 5 custom field');
+      const notCategoryPlugin = screen.queryByText('Config Sync');
+      expect(categoryPlugin).toBeVisible();
+      expect(notCategoryPlugin).toEqual(null);
+    });
+
+    it('filters multiple collection options', async () => {
+      userEvent.click(screen.getByRole('button', { name: /filters/i }));
+      userEvent.click(screen.getByRole('button', { name: 'No collections selected' }));
+      userEvent.click(screen.getByRole('option', { name: `Made by Strapi (13)` }));
+
+      await waitForReload();
+
+      userEvent.click(screen.getByRole('button', { name: /filters/i }));
+      userEvent.click(screen.getByRole('button', { name: `1 collection selected Made by Strapi` }));
+      userEvent.click(screen.getByRole('option', { name: `Verified (29)` }));
+
+      await waitForReload();
+
+      const madeByStrapiTag = screen.getByRole('button', { name: 'Made by Strapi' });
+      const verifiedTag = screen.getByRole('button', { name: 'Verified' });
+      expect(madeByStrapiTag).toBeVisible();
+      expect(verifiedTag).toBeVisible();
+      expect(screen.getAllByTestId('npm-package-card').length).toEqual(3);
+      expect(screen.getByText('Gatsby Preview')).toBeVisible();
+      expect(screen.getByText('Config Sync')).toBeVisible();
+      expect(screen.queryByText('Comments')).toEqual(null);
+    });
+
+    it('filters multiple category options', async () => {
+      userEvent.click(screen.getByRole('button', { name: /filters/i }));
+      userEvent.click(screen.getByRole('button', { name: 'No categories selected' }));
+      userEvent.click(screen.getByRole('option', { name: `Custom fields (4)` }));
+
+      await waitForReload();
+
+      userEvent.click(screen.getByRole('button', { name: /filters/i }));
+      userEvent.click(screen.getByRole('button', { name: `1 category selected Custom fields` }));
+      userEvent.click(screen.getByRole('option', { name: `Monitoring (1)` }));
+
+      await waitForReload();
+
+      const customFieldsTag = screen.getByRole('button', { name: 'Custom fields' });
+      const monitoringTag = screen.getByRole('button', { name: 'Monitoring' });
+      expect(customFieldsTag).toBeVisible();
+      expect(monitoringTag).toBeVisible();
+      expect(screen.getAllByTestId('npm-package-card').length).toEqual(3);
+      expect(screen.getByText('CKEditor 5 custom field')).toBeVisible();
+      expect(screen.getByText('Sentry')).toBeVisible();
+      expect(screen.queryByText('Comments')).toEqual(null);
+    });
+
+    it('removes a filter option tag', async () => {
```

Comment:
ℹ️ I ran into an issue testing this in each test case where there are multiple filters applied. For some reason it asserts faster than the history is updated. I was able to use a setTimeout of 2000ms to get it working but it didn't feel right. I think the real solution is to upgrade `user-event` to v14 to benefit from async await. For now I think this test case is the way to go.

Reply:
Are you sure that awaiting on the click will actually make sure the history gets updated in time?

Reply:
No not really sorry should have specified that is just a hunch